### ISSUE TITLE
simple tile will crash wayfire while moving small dialogs

### ISSUE THREAD

**Issue by killown** [WayfireWM/wayfire#2241]
steps to reproduces, 
have a small dialog and move it around (maybe 200x200) ? try different sizes to reproduce
<details>

```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==49768==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x56b3b714d086 bp 0x7ffe68c01d40 sp 0x7ffe68c01aa0 T0)
==49768==The signal is caused by a READ memory access.
==49768==Hint: address points to the zero page.
    #0 0x56b3b714d086 in wf::pointer_t::send_leave_to_focus() ../src/core/seat/pointer.cpp:108
    #1 0x56b3b714d6c9 in wf::pointer_t::update_cursor_focus(std::shared_ptr<wf::scene::node_t>) ../src/core/seat/pointer.cpp:164
    #2 0x56b3b7159b26 in wf::pointer_t::update_cursor_position(long) ../src/core/seat/pointer.cpp:83
    #3 0x56b3b715a100 in wf::pointer_t::grab_surface(std::shared_ptr<wf::scene::node_t>) ../src/core/seat/pointer.cpp:202
    #4 0x56b3b715a100 in wf::pointer_t::grab_surface(std::shared_ptr<wf::scene::node_t>) ../src/core/seat/pointer.cpp:186
    #5 0x56b3b715bdd0 in operator() ../src/core/seat/pointer.cpp:29
    #6 0x56b3b715bdd0 in __invoke_impl<void, wf::pointer_t::pointer_t(nonstd::observer_ptr<wf::input_manager_t>, nonstd::observer_ptr<wf::seat_t>)::<lambda(wf::scene::root_node_update_signal*)>&, wf::scene::root_node_update_signal*> /usr/include/c++/13.2.1/bits/invoke.h:61
    #7 0x56b3b715bdd0 in __invoke_r<void, wf::pointer_t::pointer_t(nonstd::observer_ptr<wf::input_manager_t>, nonstd::observer_ptr<wf::seat_t>)::<lambda(wf::scene::root_node_update_signal*)>&, wf::scene::root_node_update_signal*> /usr/include/c++/13.2.1/bits/invoke.h:111
    #8 0x56b3b715bdd0 in _M_invoke /usr/include/c++/13.2.1/bits/std_function.h:290
    #9 0x56b3b70bad6e in std::function<void (void*)>::operator()(void*) const /usr/include/c++/13.2.1/bits/std_function.h:591
    #10 0x56b3b70bad6e in wf::signal::connection_t<wf::scene::root_node_update_signal>::emit(wf::scene::root_node_update_signal*) ../src/api/wayfire/signal-provider.hpp:107
    #11 0x56b3b70bad6e in wf::signal::provider_t::emit<wf::scene::root_node_update_signal>(wf::scene::root_node_update_signal*)::{lambda(wf::signal::connection_base_t*)#1}::operator()(wf::signal::connection_base_t*) const ../src/api/wayfire/signal-provider.hpp:163
    #12 0x56b3b70bad6e in void std::__invoke_impl<void, wf::signal::provider_t::emit<wf::scene::root_node_update_signal>(wf::scene::root_node_update_signal*)::{lambda(wf::signal::connection_base_t*)#1}&, wf::signal::connection_base_t*&>(std::__invoke_other, wf::signal::provider_t::emit<wf::scene::root_node_update_signal>(wf::scene::root_node_update_signal*)::{lambda(wf::signal::connection_base_t*)#1}&, wf::signal::connection_base_t*&) /usr/include/c++/13.2.1/bits/invoke.h:61
    #13 0x56b3b70bad6e in std::enable_if<is_invocable_r_v<void, wf::signal::provider_t::emit<wf::scene::root_node_update_signal>(wf::scene::root_node_update_signal*)::{lambda(wf::signal::connection_base_t*)#1}&, wf::signal::connection_base_t*&>, void>::type std::__invoke_r<void, wf::signal::provider_t::emit<wf::scene::root_node_update_signal>(wf::scene::root_node_update_signal*)::{lambda(wf::signal::connection_base_t*)#1}&, wf::signal::connection_base_t*&>(wf::signal::provider_t::emit<wf::scene::root_node_update_signal>(wf::scene::root_node_update_signal*)::{lambda(wf::signal::connection_base_t*)#1}&, wf::signal::connection_base_t*&) /usr/include/c++/13.2.1/bits/invoke.h:111
    #14 0x56b3b70bad6e in std::_Function_handler<void (wf::signal::connection_base_t*&), wf::signal::provider_t::emit<wf::scene::root_node_update_signal>(wf::scene::root_node_update_signal*)::{lambda(wf::signal::connection_base_t*)#1}>::_M_invoke(std::_Any_data const&, wf::signal::connection_base_t*&) /usr/include/c++/13.2.1/bits/std_function.h:290
    #15 0x56b3b7023ff7 in std::function<void (wf::signal::connection_base_t*&)>::operator()(wf::signal::connection_base_t*&) const /usr/include/c++/13.2.1/bits/std_function.h:591
    #16 0x56b3b7023ff7 in wf::safe_list_t<wf::signal::connection_base_t*>::for_each(std::function<void (wf::signal::connection_base_t*&)>) ../src/api/wayfire/nonstd/safe-list.hpp:78
    #17 0x56b3b70c0e12 in void wf::signal::provider_t::emit<wf::scene::root_node_update_signal>(wf::scene::root_node_update_signal*) ../src/api/wayfire/signal-provider.hpp:159
    #18 0x56b3b70c0e12 in wf::scene::update(std::shared_ptr<wf::scene::node_t>, unsigned int) ../src/core/scene.cpp:499
    #19 0x77ece311040b in wf::scene::remove_child(std::shared_ptr<wf::scene::node_t>, unsigned int) ../src/api/wayfire/scene-operations.hpp:31
    #20 0x77ece31348d3 in wf::input_grab_t::ungrab_input() ../plugins/common/wayfire/plugins/common/input-grab.hpp:163
    #21 0x77ece31348d3 in wf::tile_output_plugin_t::stop_controller(bool) ../plugins/tile/tile-plugin.cpp:115
    #22 0x77ece317afd2 in wf::tile_output_plugin_t::on_view_unmapped::{lambda(wf::view_unmapped_signal*)#1}::operator()(wf::view_unmapped_signal*) const ../plugins/tile/tile-plugin.cpp:162
    #23 0x77ece317afd2 in void std::__invoke_impl<void, wf::tile_output_plugin_t::on_view_unmapped::{lambda(wf::view_unmapped_signal*)#1}&, wf::view_unmapped_signal*>(std::__invoke_other, wf::tile_output_plugin_t::on_view_unmapped::{lambda(wf::view_unmapped_signal*)#1}&, wf::view_unmapped_signal*&&) /usr/include/c++/13.2.1/bits/invoke.h:61
    #24 0x77ece317afd2 in std::enable_if<is_invocable_r_v<void, wf::tile_output_plugin_t::on_view_unmapped::{lambda(wf::view_unmapped_signal*)#1}&, wf::view_unmapped_signal*>, void>::type std::__invoke_r<void, wf::tile_output_plugin_t::on_view_unmapped::{lambda(wf::view_unmapped_signal*)#1}&, wf::view_unmapped_signal*>(wf::tile_output_plugin_t::on_view_unmapped::{lambda(wf::view_unmapped_signal*)#1}&, wf::view_unmapped_signal*&&) /usr/include/c++/13.2.1/bits/invoke.h:111
    #25 0x77ece317afd2 in std::_Function_handler<void (wf::view_unmapped_signal*), wf::tile_output_plugin_t::on_view_unmapped::{lambda(wf::view_unmapped_signal*)#1}>::_M_invoke(std::_Any_data const&, wf::view_unmapped_signal*&&) /usr/include/c++/13.2.1/bits/std_function.h:290
    #26 0x56b3b71c7d2e in std::function<void (void*)>::operator()(void*) const /usr/include/c++/13.2.1/bits/std_function.h:591
    #27 0x56b3b71c7d2e in wf::signal::connection_t<wf::view_unmapped_signal>::emit(wf::view_unmapped_signal*) ../src/api/wayfire/signal-provider.hpp:107
    #28 0x56b3b71c7d2e in wf::signal::provider_t::emit<wf::view_unmapped_signal>(wf::view_unmapped_signal*)::{lambda(wf::signal::connection_base_t*)#1}::operator()(wf::signal::connection_base_t*) const ../src/api/wayfire/signal-provider.hpp:163
    #29 0x56b3b71c7d2e in void std::__invoke_impl<void, wf::signal::provider_t::emit<wf::view_unmapped_signal>(wf::view_unmapped_signal*)::{lambda(wf::signal::connection_base_t*)#1}&, wf::signal::connection_base_t*&>(std::__invoke_other, wf::signal::provider_t::emit<wf::view_unmapped_signal>(wf::view_unmapped_signal*)::{lambda(wf::signal::connection_base_t*)#1}&, wf::signal::connection_base_t*&) /usr/include/c++/13.2.1/bits/invoke.h:61
    #30 0x56b3b71c7d2e in std::enable_if<is_invocable_r_v<void, wf::signal::provider_t::emit<wf::view_unmapped_signal>(wf::view_unmapped_signal*)::{lambda(wf::signal::connection_base_t*)#1}&, wf::signal::connection_base_t*&>, void>::type std::__invoke_r<void, wf::signal::provider_t::emit<wf::view_unmapped_signal>(wf::view_unmapped_signal*)::{lambda(wf::signal::connection_base_t*)#1}&, wf::signal::connection_base_t*&>(wf::signal::provider_t::emit<wf::view_unmapped_signal>(wf::view_unmapped_signal*)::{lambda(wf::signal::connection_base_t*)#1}&, wf::signal::connection_base_t*&) /usr/include/c++/13.2.1/bits/invoke.h:111
    #31 0x56b3b71c7d2e in std::_Function_handler<void (wf::signal::connection_base_t*&), wf::signal::provider_t::emit<wf::view_unmapped_signal>(wf::view_unmapped_signal*)::{lambda(wf::signal::connection_base_t*)#1}>::_M_invoke(std::_Any_data const&, wf::signal::connection_base_t*&) /usr/include/c++/13.2.1/bits/std_function.h:290
    #32 0x56b3b7023ff7 in std::function<void (wf::signal::connection_base_t*&)>::operator()(wf::signal::connection_base_t*&) const /usr/include/c++/13.2.1/bits/std_function.h:591
    #33 0x56b3b7023ff7 in wf::safe_list_t<wf::signal::connection_base_t*>::for_each(std::function<void (wf::signal::connection_base_t*&)>) ../src/api/wayfire/nonstd/safe-list.hpp:78
    #34 0x56b3b71d3dde in void wf::signal::provider_t::emit<wf::view_unmapped_signal>(wf::view_unmapped_signal*) ../src/api/wayfire/signal-provider.hpp:159
    #35 0x56b3b71d3dde in wf::view_interface_t::emit_view_unmap() ../src/view/view-impl.cpp:65
    #36 0x77ece5f5f0e5 in wf::preview_indication_t::update_animation() ../plugins/common/wayfire/plugins/common/preview-indication.hpp:157
    #37 0x77ece5f5f0e5 in wf::preview_indication_t::preview_indication_t(wlr_box, wf::output_t*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda()#1}::operator()() const ../plugins/common/wayfire/plugins/common/preview-indication.hpp:61
    #38 0x77ece5f5f0e5 in void std::__invoke_impl<void, wf::preview_indication_t::preview_indication_t(wlr_box, wf::output_t*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda()#1}&>(std::__invoke_other, wf::preview_indication_t::preview_indication_t(wlr_box, wf::output_t*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda()#1}&) /usr/include/c++/13.2.1/bits/invoke.h:61
    #39 0x77ece5f5f0e5 in std::enable_if<is_invocable_r_v<void, wf::preview_indication_t::preview_indication_t(wlr_box, wf::output_t*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda()#1}&>, void>::type std::__invoke_r<void, wf::preview_indication_t::preview_indication_t(wlr_box, wf::output_t*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda()#1}&>(wf::preview_indication_t::preview_indication_t(wlr_box, wf::output_t*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda()#1}&) /usr/include/c++/13.2.1/bits/invoke.h:111
    #40 0x77ece5f5f0e5 in std::_Function_handler<void (), wf::preview_indication_t::preview_indication_t(wlr_box, wf::output_t*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda()#1}>::_M_invoke(std::_Any_data const&) /usr/include/c++/13.2.1/bits/std_function.h:290
    #41 0x56b3b72bd744 in std::function<void (wf::signal::connection_base_t*&)>::operator()(wf::signal::connection_base_t*&) const /usr/include/c++/13.2.1/bits/std_function.h:591
    #42 0x56b3b72bd744 in wf::safe_list_t<std::function<void ()>*>::for_each(std::function<void (std::function<void ()>*&)>) ../src/api/wayfire/nonstd/safe-list.hpp:78
    #43 0x56b3b72bd744 in wf::effect_hook_manager_t::run_effects(wf::output_effect_type_t) ../src/output/render-manager.cpp:485
    #44 0x56b3b72c7457 in wf::render_manager::impl::paint() ../src/output/render-manager.cpp:1086
    #45 0x56b3b72d2bee in wf::render_manager::impl::impl(wf::output_t*)::{lambda(void*)#1}::operator()(void*) const ../src/output/render-manager.cpp:934
    #46 0x56b3b72d2bee in void std::__invoke_impl<void, wf::render_manager::impl::impl(wf::output_t*)::{lambda(void*)#1}&, void*>(std::__invoke_other, wf::render_manager::impl::impl(wf::output_t*)::{lambda(void*)#1}&, void*&&) /usr/include/c++/13.2.1/bits/invoke.h:61
    #47 0x56b3b72d2bee in std::enable_if<is_invocable_r_v<void, wf::render_manager::impl::impl(wf::output_t*)::{lambda(void*)#1}&, void*>, void>::type std::__invoke_r<void, wf::render_manager::impl::impl(wf::output_t*)::{lambda(void*)#1}&, void*>(wf::render_manager::impl::impl(wf::output_t*)::{lambda(void*)#1}&, void*&&) /usr/include/c++/13.2.1/bits/invoke.h:111
    #48 0x56b3b72d2bee in std::_Function_handler<void (void*), wf::render_manager::impl::impl(wf::output_t*)::{lambda(void*)#1}>::_M_invoke(std::_Any_data const&, void*&&) /usr/include/c++/13.2.1/bits/std_function.h:290
    #49 0x56b3b7003319 in std::function<void (void*)>::operator()(void*) const /usr/include/c++/13.2.1/bits/std_function.h:591
    #50 0x56b3b7003319 in wf::wl_listener_wrapper::emit(void*) ../src/wl-listener-wrapper.tpp:57
    #51 0x56b3b7003319 in handle_wrapped_listener ../src/wl-listener-wrapper.tpp:10
    #52 0x77ed0080801d in wl_signal_emit_mutable (/usr/lib/libwayland-server.so.0+0xa01d) (BuildId: d943a6a6069d1b5293dad7c842d26ce407ebdd19)
    #53 0x77ecffd14564 in handle_page_flip ../subprojects/wlroots/backend/drm/drm.c:1762
    #54 0x77ecffb35906 in drmHandleEvent (/usr/lib/libdrm.so.2+0xe906) (BuildId: 37b8c59fab0dabcaaf47293ff8e2b9fa19096a0a)
    #55 0x77ecffd03e12 in handle_drm_event ../subprojects/wlroots/backend/drm/drm.c:1774
    #56 0x77ed00809ae1 in wl_event_loop_dispatch (/usr/lib/libwayland-server.so.0+0xbae1) (BuildId: d943a6a6069d1b5293dad7c842d26ce407ebdd19)
    #57 0x77ed0080a2d6 in wl_display_run (/usr/lib/libwayland-server.so.0+0xc2d6) (BuildId: d943a6a6069d1b5293dad7c842d26ce407ebdd19)
    #58 0x56b3b6ffe19e in main ../src/main.cpp:443
    #59 0x77ecff557ccf  (/usr/lib/libc.so.6+0x25ccf) (BuildId: c0caa0b7709d3369ee575fcd7d7d0b0fc48733af)
    #60 0x77ecff557d89 in __libc_start_main (/usr/lib/libc.so.6+0x25d89) (BuildId: c0caa0b7709d3369ee575fcd7d7d0b0fc48733af)
    #61 0x56b3b7003074 in _start (/usr/bin/wayfire+0xf9074) (BuildId: 40bd55a001c4dd94ce4a9d1c5a60fdd08b66cbd1)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV ../src/core/seat/pointer.cpp:108 in wf::pointer_t::send_leave_to_focus()
==49768==ABORTING
(EE) failed to read Wayland events: Broken pipe
```

**Comment by killown** [comment]
The reliable way to reproduce the issue is do exactly the same way I am doing, you need obs, 
1 - open 3 views, one view is alacritty for example
2 - open obs from alacritty and close the terminal.
3 - open obs again, now you have a dialog with options for normal or secure mode
4 - drag and move the dialog around the output until it crash, no need to drop

**Comment by ammen99** [comment]
@killown I reproduced the bug once in another way but failed to figure out how to repro reliably (and I don't want to install OBS) in any case I think by looking at the stacktrace I have an idea what is wrong and fixed a related bug - can you try with #2268, is the original bug reproducible?

**Comment by killown** [comment]
> @killown I reproduced the bug once in another way but failed to figure out how to repro reliably (and I don't want to install OBS) in any case I think by looking at the stacktrace I have an idea what is wrong and fixed a related bug - can you try with #2268, is the original bug reproducible?

Fixed, thanks
